File: backend/WMS-mini/src/WMS.Domain/Entities/Product.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using WMS.Domain.Common;

namespace WMS.Domain.Entities;

public class Product : BaseAuditableEntity
{
    [MaxLength(50)]
    public string Sku { get; set; } = string.Empty;

    [MaxLength(200)]
    public string Name { get; set; } = string.Empty;

    public Guid CategoryId { get; set; }

    [Required]
    public Category Category { get; set; } = null!;

    [MaxLength(50)]
    public string? Unit { get; set; }

    [Column(TypeName = "decimal(18,2)")]
    public decimal Price { get; set; }

    [MaxLength(100)]
    public string? Dimension { get; set; }
}

File: frontend/src/types/product.ts
export interface ProductDto {
  id: string
  sku: string
  name: string
  categoryId: string
  unit?: string
  price: number
  dimension?: string
  imageUrl?: string
}

export interface CreateProductDto {
  sku: string
  name: string
  categoryId: string
  unit?: string
  price: number
  dimension?: string
}